conda_libmamba_solver/solver.py

```diff
@@ -741,6 +755,24 @@ def _check_spec_compat(self, match_spec):
                 f"You can only use {supported}, but you tried to use "
                 f"{tuple(unsupported_but_set)}.",
             )
+        if match_spec.get_raw_value("channel") == "defaults":
+            # !!! Temporary !!!
+            # Apply workaround for defaults::pkg-name specs.
+            # We need to replace it with the actual channel name (main, msys2, r)
+            # Instead of searching in the index, we apply a simple heuristic:
+            # - R packages are [_]r-*, mro-*, rpy or rstudio
+            # - Msys2 packages are m2-*, m2w64-*, or msys2-*
+            # - Everything else is in main
+            name = match_spec.name.lower()
+            if name in ("r", "rpy2", "rstudio") or name.startswith(("r-", "_r-", "mro-")):
+                channel = "pkgs/r"
+            elif name.startswith(("m2-", "m2w64-", "msys2-")):
+                channel = "pkgs/msys2"
```

Comment:
Yikes, this hard-coding is worrying, can we get that list from somewhere?

Reply:
Not without querying the `index` and summarising ourselves. This is just a temporary patch to enable `defaults::*` functionality, which is broken on `libmamba` right now. I am hoping that this is fixed by the next release so we can remove this. I don't think `msys2` and `r` see too many changes these days anyway, and the number of people doing `defaults::r` should be very small (they would just do `-c defaults r` if anything).

Reply:
For the record, the upstream PR is https://github.com/conda/conda/pull/12555